mobot/mobot_client/payments.py

```diff
@@ -0,0 +1,186 @@
+        # I'm assuming there will only be one receiver receipt (not including change tx out).
+        if len(receiver_receipts) > 1:
+            raise ValueError(
+                "Found more than one txout for this chat bot-initiated transaction."
+            )
+        return receiver_receipts[0]
+
+    def get_unspent_pmob(self):
+        account_amount_response = self.mcc.get_balance_for_account(self.account_id)
+        unspent_pmob = int(account_amount_response["unspent_pmob"])
+        return unspent_pmob
+
+    def get_minimum_fee_pmob(self):
+        return self.minimum_fee_pmob
+
+    def handle_item_payment(self, source, customer, amount_paid_mob, drop_session):
+        item_cost_mob = mc.pmob2mob(drop_session.drop.item.price_in_pmob)
+
+        if amount_paid_mob < item_cost_mob:
+            if mc.mob2pmob(amount_paid_mob) > self.minimum_fee_pmob:
+                self.messenger.log_and_send_message(
+                    customer,
+                    source,
+                    f"Not enough MOB, sending back {amount_paid_mob.normalize()} (minus network fees)",
+                )
+                self.send_mob_to_customer(customer, source, amount_paid_mob, False)
+            else:
+                self.messenger.log_and_send_message(
+                    customer,
+                    source,
+                    "Not enough MOB, unable to refund since it is less than the network fee",
+                )
+            return
+
+        if (
+                mc.mob2pmob(amount_paid_mob)
+                > mc.mob2pmob(item_cost_mob) + self.minimum_fee_pmob
+        ):
+            excess = amount_paid_mob - item_cost_mob
+            self.messenger.log_and_send_message(
+                customer,
+                source,
+                f"Sent too much MOB, sending back excess of {excess.normalize()} (minus network fees)",
+            )
+            self.send_mob_to_customer(customer, source, excess, False)
+
+        available_options = []
+        skus = Sku.objects.filter(item=drop_session.drop.item)
+
+        for sku in skus:
+            number_ordered = Order.objects.filter(sku=sku).count()
+            if number_ordered < sku.quantity:
+                available_options.append(sku)
+
+        if len(available_options) == 0:
+            self.messenger.log_and_send_message(
+                customer,
+                source,
+                "Uh oh! Looks like we're all out of stock, sorry! Refunding your payment now :)",
+            )
+            self.send_mob_to_customer(customer, source, item_cost_mob, True)
+            drop_session.state = ItemSessionState.REFUNDED.value
+            drop_session.save()
+            return
+
+        message_to_send = (
+            "What size would you like? We have the following available options:\n\n"
+        )
+        for option in available_options:
+            message_to_send += f" - {option.identifier}\n"
```

Comment:
Isn't this the purpose of https://github.com/mobilecoinofficial/mobot/blob/2f369f32ec84b2882bff0387c20811fe9b3c20d4/mobot/mobot_client/chat_strings.py#L160 ?